javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -186,7 +186,7 @@ final class Matcher private[regex] (
 
   def groupCount(): Int = Matcher.getGroupCount(lastMatch, pattern())
 
-  def start(): Int = ensureLastMatch.index
+  def start(): Int = ensureLastMatch.index + regionStart0
```

Comment:
```suggestion
  def start(): Int = ensureLastMatch.index + regionStart()
```
like you do further down for `toMatchResult()`? Or use `regionStart0` also in `toMatchResult`? It's a bit inconsistent, at the moment, which led me to believe that there was an actual difference.